utils/s3/client.cc

```diff
@@ -223,6 +232,64 @@ future<client::stats> client::get_object_stats(sstring object_name) {
     co_return st;
 }
 
+future<client::TagSet> client::get_object_tagging(sstring object_name) {
+    // see https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectTagging.html
+    auto req = http::request::make("GET", _host, fmt::format("{}?tagging", object_name));
+    s3l.trace("GET {} tagging", object_name);
+    TagSet tagging;
+    co_await make_request(std::move(req),
+                          [&tagging] (const http::reply& reply, input_stream<char>&& in) mutable -> future<> {
+        auto& tagset = tagging;
+        auto input = std::move(in);
+        auto body = co_await util::read_entire_stream_contiguous(input);
+        auto doc = std::make_unique<rapidxml::xml_document<>>();
+        try {
+            doc->parse<0>(body.data());
+        } catch (const rapidxml::parse_error& e) {
+            s3l.warn("cannnot parse tagging response: {}", e.what());
+            throw std::runtime_error("cannot parse tagging response");
+        }
+        auto tagset_node = doc->first_node("Tagging")->first_node("TagSet");
```

Comment:
I'm not familiar with this "rapidxml" library, does first_node() throw an exception if there is no such node? Will the exception be something reasonable that the user/debugger/etc. be able to understand?

Reply:
good point. will early return in this case.

Reply:
see also https://rapidxml.sourceforge.net/manual.html#classrapidxml_1_1xml__node_7823e36687669e59c2afdf66334ef35a_17823e36687669e59c2afdf66334ef35a and https://github.com/discord/rapidxml/blob/2ae4b2888165a393dfb6382168825fddf00c27b9/rapidxml.hpp#L936-L946